lib/src/log_handler.dart

```diff
@@ -0,0 +1,250 @@
+  ///                 "time_ns": 345
+  ///             }
+  ///         }
+  ///     ],
+  ///     "user_properties": {
+  ///         "session_id": {
+  ///             "value": 1675193534342
+  ///         },
+  ///         "branch": {
+  ///             "value": "ey-test-branch"
+  ///         },
+  ///         "host": {
+  ///             "value": "macOS"
+  ///         },
+  ///         "flutter_version": {
+  ///             "value": "Flutter 3.6.0-7.0.pre.47"
+  ///         },
+  ///         "dart_version": {
+  ///             "value": "Dart 2.19.0"
+  ///         },
+  ///         "tool": {
+  ///             "value": "flutter-tools"
+  ///         },
+  ///         "local_time": {
+  ///             "value": "2023-01-31 14:32:14.592898"
+  ///         }
+  ///     }
+  /// }
+  /// ```
+  static LogItem? fromRecord(Map<String, Object?> record) {
+    if (!record.containsKey('user_properties')) return null;
+
+    // Using a try/except here to parse out the fields if possible,
+    // if not, it will quietly return null and won't get processed
+    // downstream
+    try {
+      // Parse the data out of the `user_properties` value
+      final Map<String, Object?> userProps =
+          record['user_properties'] as Map<String, Object?>;
+
+      // Parse out the values from the top level key = 'user_properties`
+      final int? sessionId =
+          (userProps['session_id']! as Map<String, Object?>)['value'] as int?;
+      final String? branch =
+          (userProps['branch']! as Map<String, Object?>)['value'] as String?;
+      final String? host =
+          (userProps['host']! as Map<String, Object?>)['value'] as String?;
+      final String? flutterVersion = (userProps['flutter_version']!
+          as Map<String, Object?>)['value'] as String?;
+      final String? dartVersion = (userProps['dart_version']!
+          as Map<String, Object?>)['value'] as String?;
+      final String? tool =
+          (userProps['tool']! as Map<String, Object?>)['value'] as String?;
+      final String? localTimeString = (userProps['local_time']!
+          as Map<String, Object?>)['value'] as String?;
+
+      // If any of the above values are null, return null since that
+      // indicates the record is malformed
+      final List<Object?> values = <Object?>[
+        sessionId,
+        branch,
+        host,
+        flutterVersion,
+        dartVersion,
+        tool,
+        localTimeString,
+      ];
+      for (Object? value in values) {
+        if (value == null) return null;
+      }
+
+      // Parse the local time from the string extracted
+      final DateTime localTime = DateTime.parse(localTimeString!);
+
+      return LogItem(
+        sessionId: sessionId!,
+        branch: branch!,
+        host: host!,
+        flutterVersion: flutterVersion!,
+        dartVersion: dartVersion!,
+        tool: tool!,
+        localTime: localTime,
+      );
+    } on Exception {
+      return null;
+    }
+  }
```

Comment:
@christopherfujino I have refactored with specific casting for each value and also checked if any of them are null in a for loop